lib/symboldatabase.h

```diff
@@ -1160,11 +1161,24 @@ class CPPCHECKLIB ValueType {
     const Token *containerTypeToken;      ///< The container type token. the template argument token that defines the container element type.
     std::string originalTypeName;         ///< original type name as written in the source code. eg. this might be "uint8_t" when type is CHAR.
 
+    bool isPointer() const {
+        return pointerDepth > 0 && !reference;
```

Comment:
in my humble opinion a `X*&` is a pointer.

Reply:
I guess it's just a question about naming? The name `isNonReferencePointer()` would be more clear but it is rather long..